Essay_Analysis_Tool/Interface/ILoggerView.cs

```diff
@@ -0,0 +1,26 @@
+﻿using Essay_Analysis_Tool.Business;
+using FastColoredTextBoxNS;
+
+namespace Essay_Analysis_Tool.Interface
+{
+    public interface ILoggerView
+    {
+        /// <summary>
+        /// Appends the given text to the FastColoredTextBox instance.
+        /// </summary>
+        /// <param name="text">Message text.</param>
+        /// <param name="style">Style of message.</param>
+        void AppendText(string text, Style style);
```

Comment:
I would change this method visibility private and i would make 3 public method instead:

- void Info(string message);
- void Warn(string message);
- void Error(string message);

With this implementation the `LoggerMessageType` could be deleted and i would strongly advise to move the `TextStyle` implementation to the View because that is not business logic that is UI.

Reply:
This is what we are going to do, however, I still have to figure out how abstracting it out to a service class is going to work. Because the service class will have to have access to the either the view or the controller. I'm not really sure which one.

Reply:
If i were You i would refactor the code to this (pseudocode):

``` c#
class LogService
{
    private List<LogEntry> _entries;
    eventhandler ItemAdded; // event will pass the LogEntry to all subscribed members
    void Add(LogEntry entry)
    {
        _entries.Add(entry);
        // invoke the eventhandler to notfiy everybody who subscribed to this event
    }
}

abstract class LogEntry
{
    public string Message { get; }
    public DateTime AddedAt { get; }
}

class InfoLogEntry : LogEntry
{
}

class WarnLogEntry : LogEntry
{
}

class ErrorLogEntry : LogEntry
{
}

class LogContoller
{
    public LogController(LogService service)
    {
        service.ItemAdded += UpdateView();
    }

    private void UpdateView(...) { view. }

    ~LogController()
    {
        service.ItemAdded -= UpdateView();
    }
}

public partial class LoggerView : Form
{
    public void Add(InfoLogEntry entry)
    {
        Append(entry, infoStyle);
    }

    public void Add(WarnLogEntry entry)
    {
        Append(entry, infoStyle);
    }

    public void Add(ErrorLogEntry entry)
    {
        Append(entry, infoStyle);
    }
    
    private void Append(entry, TextStyle style)
    {
        // console.AppendText($"{entry.AddedAt}: {entry.Message}", style);
        console.AppendText(entry.AddedAt + ": " + entry.Message, style);
    }
}

var logService = new LogService();
var logController = new LogController(logService);


logService.Add(new InfoLogEntry("Form Initialized!"));
logService.Add(new ErrorLogEntry("System crash! ..."));
```

Reply:
I decided to remove the interfaces lol.

Reply:
I made a working example in the **[feature/architecture-logging](https://github.com/Hexman768/Notepad-Sharp/tree/feature/architecture-logging)** branch but my psudo example was wrong and i am not completely satisfied this code, it feels i force the LogEntry class thing.